reader/src/test/scala/com/expedia/www/haystack/trace/reader/unit/readers/transformers/ServerClientSpanMergeTransformerSpec.scala

```diff
@@ -195,8 +231,23 @@ class ServerClientSpanMergeTransformerSpec extends BaseUnitTestSpec with ValidTr
       spanTree.children.head.children.map(_.span) should contain allOf(underlyingSpans.apply(2), underlyingSpans.apply(3))
       spanTree.children.head.children.foreach(tree => tree.children.size shouldBe 0)
     }
+
+    it ("should not merge producer and consumer parent-child spans") {
```

Comment:
I guess before it was merging regardless of span ID? or were you getting producer and consumer with the same ID?

Reply:
We need a fix for a usecase where a singled produced message in kafka gets consumed by multiple consumer apps. The merge logic in haystack works good for a client-server span but for this kafka usecase, we can't merge the spans as there is no one consumer.

Reply:
right I mean to say usually they have different span IDs so I guess in your scenario multiple consumers are still using the same span ID (hence being merged) right?

Reply:
right.